plugins/gradle/src/main/java/com/slapin/napt/NaptGradlePlugin.kt

```diff
@@ -33,15 +33,17 @@ class NaptGradlePlugin : Plugin<Project> {
     }
 
     private fun Project.bindTriggerCreation(extension: NaptGradleExtension) {
-
         val createTrigger =
             tasks.register(
                 "createNaptTrigger",
                 CreateNaptTrigger::class.java,
             ) { task ->
-                task.mainSourceSetDir.set(layout.projectDirectory.dir("src/main"))
+                task.mainSourceSetJavaDirPath.set(
+                    layout.projectDirectory.dir("src/main/java").asFile.path
+                )
                 task.projectName.set(name)
                 task.packagePrefix.set(extension.naptTriggerPackagePrefix)
+                task.onlyIf { extension.generateNaptTrigger.getOrElse(true) }
```

Comment:
I think you want to use `task.inouts.upToDateWhen { }`

Reply:
Not sure about that, it feels more natural to set task to be skipped in this case rather than saying that its outputs can be reused as there's no outputs.

Reply:
ah, I see what you mean 👍